### DISCUSSION TITLE
Opening a new pane directly embeded

### DISCUSSION THREAD

**Discussion by cosminadrianpopescu** [zellij-org/zellij#1379] [Ideas]
Following [this](https://github.com/zellij-org/zellij/issues/1353), now I can have a shortcut like this:

```yaml
        - action: [DumpScreen: /tmp/log1.sh, Run: {cmd: /opt/nvim.appimage, args: [/tmp/log1.sh, +normal G]}, SwitchToMode: Locked]
          key: [Char: '/']
```

The issue with the `Run` command is that if I dump the buffer in a floating panel, it will open a floating pane with the buffer. 

I have some proposals: 

1. We can add a flag that should specify that I want the popup opened embeded (in this case, the floating panels would be hidden and a new embeded pane would be opened)
or
2. I was wondering why the floating panel does not have a full screen option. That would be amazing. And then I would propose a flag to Run command to open the popup directly in fullscreen

So, what I'm proposing is this:

* Add fullscreen option to the floating panels (is there any technical issues that these cannot be full screened)
* Add a new open to the `Run` command to be able to open the new popup directly in full screen mode (and maybe another flag with embed). 

What do you think?

**Comment by imsnif** [comment]
This all sounds great and I'd be on board with implementing everything.

> We can add a flag that should specify that I want the popup opened embeded (in this case, the floating panels would be hidden and a new embeded pane would be opened)

That would be great. Something like ` Run: { ... floating: true}` or `Run: { ... embedded: true}`

> I was wondering why the floating panel does not have a full screen option. That would be amazing. And then I would propose a flag to Run command to open the popup directly in fullscreen

Yes! I'm missing this too in my day-to-day work. There aren't technical difficulties, just some internal-logic difficulties. Namely implementing something like this for floating panes: https://github.com/zellij-org/zellij/blob/main/zellij-server/src/panes/tiled_panes/mod.rs#L70
We would have to use it in a similar way to make sure that when we render the floating panes, we don't render any panes contained in that HashSet (both for performance and so that we don't have to start dancing around with z-indices to not render anything above the maximized pane).
Maybe other stuff I'm not thinking about, but I guess this would be the main thing.
